src/llmcompressor/utils/helpers.py

```diff
@@ -1080,3 +1084,32 @@ def __enter__(self):
 
     def __exit__(self, _exc_type, _exc_val, _exc_tb):
         self.config.use_cache = self.restore_value
+
+
+@contextlib.contextmanager
+def DisableQuantization(model: torch.nn.Module):
+    """
+    Disable quantization from QuantizationModifier
+    """
+    model.apply(disable_quantization)
+    yield
+    model.apply(enable_quantization)
+
+
+@contextlib.contextmanager
+def calibration_forward_context(model: torch.nn.Module):
+    """
+    Context in which all calibration forward passes should occur.
+
+    - Remove gradient calculations
+    - Disable the KV cache
+    - Disable quantization from QuantizationModifier
+    """
+    model.eval()
+
+    with (
+        torch.no_grad(),
+        DisableKVCache(model),
```

Comment:
do we actually disable the kv cache, anywhere outside of gptq?

Reply:
No, but technically we should be disabling the kv cache for all calibration forward passes (pipelines), as calibration only consists of prefill and populating the kvcache only costs performance and memory.

Reply:
Yeah I was gonna suggest adding this to SparseGPTQ but can do later